src/unit/test_listpack.c

```diff
@@ -0,0 +1,1437 @@
+    return lp;
+}
+
+static long long usec(void) {
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    return (((long long)tv.tv_sec) * 1000000) + tv.tv_usec;
+}
+
+static void stress(int pos, int num, int maxsize, int dnum) {
+    int i, j, k;
+    unsigned char *lp;
+    char posstr[2][5] = {"HEAD", "TAIL"};
+    long long start;
+    for (i = 0; i < maxsize; i += dnum) {
+        lp = lpNew(0);
+        for (j = 0; j < i; j++) {
+            lp = lpAppend(lp, (unsigned char *)"quux", 4);
+        }
+
+        /* Do num times a push+pop from pos */
+        start = usec();
+        for (k = 0; k < num; k++) {
+            if (pos == 0) {
+                lp = lpPrepend(lp, (unsigned char *)"quux", 4);
+            } else {
+                lp = lpAppend(lp, (unsigned char *)"quux", 4);
+            }
+            lp = lpDelete(lp, lpFirst(lp), NULL);
+        }
+        printf("List size: %8d, bytes: %8zu, %dx push+pop (%s): %6lld usec\n", i, lpBytes(lp), num, posstr[pos],
+               usec() - start);
+        lpFree(lp);
+    }
+}
+
+/* simple wrapper to used in functions whose return type is not int */
+static inline int _test_assert(int c) {
+    TEST_ASSERT(c);
+    return 0;
+}
+static unsigned char *pop(unsigned char *lp, int where, void *expected) {
+    unsigned char *p, *vstr;
+    int64_t vlen;
+
+    p = lpSeek(lp, where == 0 ? 0 : -1);
+    vstr = lpGet(p, &vlen, NULL);
+
+    if (vstr) {
+        _test_assert(strcmp((const char *)vstr, (const char *)expected));
+    } else {
+        _test_assert(vlen == *(int64_t *)expected);
+    }
+
+    return lpDelete(lp, p, &p);
+}
+
+static int randstring(char *target, unsigned int min, unsigned int max) {
+    int p = 0;
+    int len = min + rand() % (max - min + 1);
+    int minval, maxval;
+    switch (rand() % 3) {
+    case 0:
+        minval = 0;
+        maxval = 255;
+        break;
+    case 1:
+        minval = 48;
+        maxval = 122;
+        break;
+    case 2:
+        minval = 48;
+        maxval = 52;
+        break;
+    default: TEST_ASSERT(NULL);
+    }
+
+    while (p < len) target[p++] = minval + rand() % (maxval - minval + 1);
+    return len;
+}
+
+static int verifyEntry(unsigned char *p, unsigned char *s, size_t slen) {
+    TEST_ASSERT(lpCompare(p, s, slen));
+    return 0;
+}
+
+static int lpValidation(unsigned char *p, unsigned int head_count, void *userdata) {
+    UNUSED(p);
+    UNUSED(head_count);
+
+    int ret;
+    long *count = userdata;
+    ret = lpCompare(p, (unsigned char *)mixlist[*count], strlen(mixlist[*count]));
+    (*count)++;
+    return ret;
+}
+
+int test_listpackCreateIntList(int argc, char **argv, int flags) {
+    /* Create int list */
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    unsigned char *lp;
+
+    lp = createIntList();
+    TEST_ASSERT(lpLength(lp) == 6);
+    lpFree(lp);
+
+    return 0;
+}
+
+
+int test_listpackCreateList(int argc, char **argv, int flags) {
```

Comment:
I see there are two line breaks in some places here, can you remove it?

Reply:
No problem. done.